compiler/qsc_rir/src/passes/ssa_transform.rs

```diff
@@ -239,17 +239,25 @@ impl VariableMapper for FxHashMap<VariableId, Operand> {
     }
 
     fn to_operand(&self, var: Variable) -> Operand {
-        self.get(&var.variable_id)
-            .copied()
-            .unwrap_or(Operand::Variable(var))
+        let mut var = var;
+        while let Some(operand) = self.get(&var.variable_id) {
+            if let Operand::Variable(new_var) = operand {
+                var = *new_var;
+            } else {
+                return *operand;
+            }
+        }
+        Operand::Variable(var)
     }
 
     fn to_variable(&self, var: Variable) -> Variable {
-        self.get(&var.variable_id)
-            .copied()
-            .map_or(var, |operand| match operand {
-                Operand::Literal(_) => panic!("literal not supported in this context"),
-                Operand::Variable(var) => var,
-            })
+        let mut var = var;
```

Comment:
nit: this looks like a `get_originating_variable` method or `get_original_variable` method. The name `to_variable` is a bit confusing because it implies a conversion, but the method is not doing that exactly.

Reply:
It seems like the previous implementation was a more straight-forward conversion, but that is no longer the case.

Reply:
That's a good point... I think this can be cleaned up with a different pattern.

Reply:
I simplified this to be more like the `impl` on `Operand` above, and I think it's more clear that it's a mapping that is occurring using the map passed as argument.